include/openenclave/internal/plugin.h

```diff
@@ -0,0 +1,465 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+/**
+ * @file plugin.h
+ *
+ * This file defines the programming interface for developing
+ * attester or verifier plugins for supporting alternative evidence formats.
+ *
+ */
+
+#ifndef _OE_ATTESTATION_PLUGIN_H
+#define _OE_ATTESTATION_PLUGIN_H
+
+#ifdef OE_BUILD_ENCLAVE
+// The attester related definitions are only available for the enclave.
+#include <openenclave/attestation/attester.h>
+#endif
+#include <openenclave/attestation/verifier.h>
+#include <openenclave/bits/report.h>
+#include <openenclave/bits/result.h>
+#include <openenclave/bits/types.h>
+
+OE_EXTERNC_BEGIN
+
+/**
+ * Struct that defines the base structure of each attestation role plugin.
+ * Each attestation role will have an UUID to indicate what evidence format
+ * is supported and have functions for registering/unregistering the plugin.
+ * Each attestation role will also define the require function for their
+ * specific role (i.e. `get_evidence` for the attester and `verifiy_evidence`
+ * for the verifier).
+ */
+typedef struct _oe_attestation_role oe_attestation_role_t;
+struct _oe_attestation_role
+{
+    /**
+     * The UUID for the attestation role.
+     */
+    oe_uuid_t format_id;
+
+    /**
+     * The function that gets executed when the attestation role is registered.
```

Comment:
```suggestion
     * The function that gets executed when the plugin is registered.
```

Rationale: "Verifier" (or "Attester", etc.) is the role... plugins are what get registered.

Reply:
Thanks!